src/mqt/predictor/rl/Predictor.py

```diff
@@ -129,118 +215,20 @@ def evaluate_sample_circuit_using(self, file):
-                "TKET Parallelism",
-                "RL Parallelism",
-                "Qiskit O3 Crit Depth",
-                "TKET Crit Depth",
-                "RL Crit Depth",
-                "Qiskit O3 Time",
-                "TKET Time",
-                "RL Fid Time",
-                "RL Parallelism Time",
-                "RL Crit Depth Time",
-                "RL Fid Crit Depth",
-                "RL Fid Parallelism",
-                "RL Crit Depth Fid",
-                "RL Crit Depth Parallelism",
-                "RL Parallelism Fid",
-                "RL Parallelism Crit Depth",
-            )
-        )
-
-        results = Parallel(n_jobs=-1, verbose=3, backend="threading")(
-            delayed(self.evaluate_sample_circuit)(file)
-            for file in list(rl.helper.get_path_training_circuits().glob("*.qasm"))
-        )
-        for res in results:
-            res_csv.append(res)
-        # print(results)
-        np.savetxt(
-            rl.helper.get_path_trained_model() / "res.csv",
-            res_csv,
-            delimiter=",",
-            fmt="%s",
-        )
-
-    def instantiate_models(self, timesteps, fid, dep, par, model_name="training"):
-        self.train_all_models(
-            timestep=timesteps,
-            verbose=2,
-            fid=fid,
-            dep=dep,
-            par=par,
-            model_name=model_name,
+        rew_fid = reward.expected_fidelity(transpiled_qc_tket, "ibm_washington")
+        rew_crit_depth = reward.crit_depth(transpiled_qc_tket)
+        rew_gate_ratio = reward.gate_ratio(transpiled_qc_tket)
+        rew_mix = reward.mix(transpiled_qc_tket, "ibm_washington")
+
+        return rl.Result(
+            benchmark=benchmark,
+            used_setup=used_setup,
+            time=duration,
+            fidelity=rew_fid,
+            depth=rew_crit_depth,
+            gate_ratio=rew_gate_ratio,
+            mix=rew_mix,
         )
 
-    def train_all_models(
-        self,
-        timestep=1000,
-        verbose=2,
-        fid=False,
-        dep=False,
-        par=False,
-        model_name="model",
-    ):
-        reward_functions = []  # ["fidelity", "depth", "two_qubit_count"]
-        if fid:
-            reward_functions.append("fidelity")
-        if dep:
-            reward_functions.append("critical_depth")
-        if par:
-            reward_functions.append("parallelism")
-
-        for rew in reward_functions:
-            print("Start training for: ", rew)
-            env = rl.PhaseOrdererEnv(reward_function=rew)
-
-            model = MaskablePPO(
-                MaskableActorCriticPolicy,
-                env,
-                verbose=verbose,
-                tensorboard_log="./" + model_name + "_" + rew,
-                gamma=0.98,
-            )
-            model.learn(total_timesteps=timestep, progress_bar=False)
-            model.save(rl.helper.get_path_trained_model() / (model_name + "_" + rew))
+    else:
```

Comment:
this else can be removed as you return at the end of each `if` branch.
Saves a line of code and an indentation level.

Reply:
done